### ISSUE TITLE
Undergraduate Research and BRAT

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/pyBRAT#295]
From a user:

>  I am looking at how beaver dams affect lateral channel migration on XXXXXX Creek here in XXXX, and we were hoping to utilize the BRAT model to help us determine suitable reaches to use for migration rate calculations (to help minimize variability in the reaches we analyze).
> 
> I attempted to utilize BRAT but seem to be having some difficulties getting it properly installed, as it appears in ArcGIS with a red ‘X’ through it. We followed all the available instructions/tips on the website but we are currently at a stand still and wondered if you had any other thoughts on what we could try.
> 
> We are under a bit of time constraint, but if we cannot use the actual BRAT toolbox, we will likely utilize the concepts that BRAT is based on to determine suitability - but we would much rather utilize the toolbox since it seems like a great method!
> 
> Thank you for your time!

**Comment by joewheaton** [comment]
Sounds like a fun senior thesis project.  Beaver dams that fail by end-cutting provide an excellent mechanism for increasing rates of lateral channel migration and kicking off bank erosion. To a lesser extent, overflow from beaver dams that leads to floodplain flows that eventually return to main channels can also claw back return flow channels that can lead to localized channel widening. Both are important mechanisms of increasing dynamism of riverscapes. However, I would caution against using some of the more classical meander migration research as a useful proxy because this is very much a structurally-forced process the timing of which is decided by beaver.  The meandering channel literature is light on examples of structurally-forced bank erosion.  Fun topic. On to your question...

**The short story is that pyBRAT IS NOT yet an easy to use GIS tool.** pyBRAT has a ton of versions specific ArcPy dependencies making it very sensitive to the version of Arc you are running and version of python. In addition, it has other python library dependencies.  This all translate into a robust method and algorithm that has gotten tons of traction in management and research circles, but NOT an easy to use Arc Script. We have other models (e.g. http://gcd.riverscapes.xyz) where we've had sustained investment, and been able to make the transition from documented, transparent open-source scripts (i.e. fragile) to more professional pieces of software that are much more user friendly and idiot proof. So far, all the interest in BRAT has mainly been in funding to produce runs where the 'output' is a 'tool' for management and we've not yet been able to convert this into an easier to use GIS tool. Frustrating....

That all said, the 'Red X' is likely something related to the version of ArcGIS and Python you are using. What version is that?

Also, if you're any good at Matlab, the old http://brat.riverscapes.xyz/Documentation/Download/#matbrat-source-code version may be of some interest. 

Finally, if you are only trying to run this for a short distance of creek, you could just manually apply the BRAT cIS (capacity inference system without fuzzy inference system part - see http://brat.riverscapes.xyz/Workshops/2018/johnday.html#brat-capacity-modelling for form and some slides to explain).  You can just qualitatively (or quantitatively) answer the questions on a reach-by-reach basis and you get an assessment as well as the inputs. 

Best of Luck,
Joe

**Comment by MeganTarmichael** [comment]
Hi Joe, 

I somehow just found your reply, I apologize for the late response! Iwas able to solve my initial problem but I think I have encountered another related to licensing (I encounter Error 000824: "The tool is not licensed" when I attempted to run the Project Builder). 

I will likely end up utilizing the BRAT cIS that you mentioned in your comment, as I only need outputs for about 16 km of stream. 

Thank you for your help, it is greatly appreciated!

Megan
